File: backend/dataset/rotate/rotate_24.py
class Solution:
    def rotate(self,nums:List[int], k:int)->None:
        if len(nums) == 0: return []
        if len(nums) == k: return nums
        if k > len(nums):
            rotations = k % len(nums)
        else:
            rotations = k
        return nums[-rotations:] + nums[:-rotations]

File: backend/CDataset/rotate/rotate_24.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
        k %= nums.size();
        while(k--){
            int temp = nums[nums.size()-1];
            for(int i = nums.size()-2; i>= 0 ;i--){
                nums[i+1] = nums[i];
            }
            nums[0] = temp;
        }   
    }
};
